src/frontends/lean/structure_cmd.cpp

```diff
@@ -499,55 +639,101 @@ struct structure_cmd_fn {
                     throw_ill_formed_parent(parent_name);
                 intro_type = instantiate(binding_body(intro_type), arg);
             }
-            size_t fmap_start = fmap.size();
-            while (is_pi(intro_type)) {
-                name fname = binding_name(intro_type);
-                fname      = rename(renames, fname);
-                expr const & ftype = binding_domain(intro_type);
-                name full_fname = parent_name + fname;
-                expr field;
-                if (auto fidx = merge(parent, fname, ftype)) {
-                    fmap.push_back(*fidx);
-                    field = m_fields[*fidx].local;
-                    if (local_info(field) != binding_info(intro_type)) {
-                        throw elaborator_exception(parent,
-                                                   sstream() << "invalid 'structure' header, field '" << fname <<
-                                                   "' has already been declared with a different binder annotation");
+            if (m_subobjects) {
+                name fname;
+                if (auto const & ref = m_parent_refs[i])
```

Comment:
I usually just copy the style of the surrounding code, so no braces if it's Leo's and always braces if it's Gabriel's ;) . But I'm also happy to adopt a consistent style in the future.